modules/auxiliary/scanner/http/iis_shortname_scanner.rb

```diff
@@ -0,0 +1,280 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Rex::Proto::Http
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'        => 'Microsoft IIS shortname vulnerability scanner',
+        'Description' => %q{
+        'The vulnerability is caused by a tilde character "~" in a GET or OPTIONS request, which
+         could allow remote attackers to diclose 8.3 filenames (short names). This is a new
+         technique for an existing bug (CVE-2009-4444), which relied on the use of the ;(semicolon)
+         character. Soroush Dalili discovered the original bug, while the new ~ technique was
+         discovered by Soroush Dalili and Ali Abbasnejad. Older IIS installations are vulnerable
+         with GET, newer installations with OPTIONS'
+        },
+        'Author'         =>
+          [
+          'MinatoTW <shaks19jais[at]gmail.com>',
```

Comment:
Soroush Dalili and Ali Abbasnejad should also be credited with the initial discovery.